### ISSUE TITLE
Error when link model

### ISSUE THREAD

**Issue by dkduyanh** [yiisoft/yii2#13805]
My business is "Each issue need to be solved by one task. So if an issue doesn't have any task, we will create the new one then link to it"

In Issue class, I declared relation "task" which store the task Id of this issue

```php
class Issue extends ActiveRecord
{
        ...
	public function getTask()
	{
		return $this->hasOne(Task::className(), ['id' => 'task_id']);
	}
}
```

The database tables:

Table Issue:

    - id
    - name
    - created_on
    - created_by
    - task_id

Table Task:

    - id
    - name
    - created_on
    - status

### What steps will reproduce the problem?

```php
$issue = Issue::findOne($issueId);
if(!$issue->task_id) /*if this issue doesn't link any task*/
{
    $task = new Task();
    $task->name = '[Issue] '.$issue->name;
    ...
    if($task->save()){
        $issue->link('task', $task); /*link new task to this issue*/
    }
}
...
```

### What is the expected result?

The task linked with issue via relation 'task'

### What do you get instead?

```
Fatal error: Call to a member function getIsNewRecord() on null 
In ...\vendor\yiisoft\yii2\db\BaseActiveRecord.php    at line 1233 
```

I debugged and found the model is null.
![problem](https://cloud.githubusercontent.com/assets/10934695/24088767/28c3f954-0d61-11e7-8be5-27859243cec6.PNG)

### Additional info

This error is only thrown for the first time. The second run (by refreshing the page) is fine because the task created at that time, so it can linked to issue successfully.

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 5.6
| Operating system | Windows 10

**Comment by dkduyanh** [comment]
I don't know why the task is null for first-time because I checked if task saved successfully before linking it to issue.

if($task->save()){
     $issue->link('task', $task); /link new task to this issue/
}

**Comment by dkduyanh** [comment]
Now, I understand the reason, the validation rule doesn't match when issue link to task model.

I think we should save the relation only instead of all attributes so we can avoid this un-clear error.

**Comment by samdark** [comment]
> Now, I understand the reason, the validation rule doesn't match when issue link to task model.

Which validation rule?

**Comment by dynasource** [comment]
from what you've posted here, I cannot see how this error could have been thrown. Its likely caused by something else in your application.

**Comment by dkduyanh** [comment]
Because each "Issue" may belongs to some categories, so I added inline-validation in the "Issue" model. The validation is "REQUIRED", this caused the Fatal error above, I'm sure because after being removed it works fine. 

To solve this problem, I added new scenario and put the "REQUIRED" rule to it, and it works fine.

My code is:

![issue](https://cloud.githubusercontent.com/assets/10934695/24131847/9e4c74f4-0e23-11e7-9095-ae020a3fc9a1.PNG)
